infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -946,6 +1001,94 @@ private AddressMetaData writeRecord(SegmentHandle fh, long address,
         return new AddressMetaData(metadata.getChecksum(), metadata.getLength(), channelOffset);
     }
 
+    long getSegment(LogData entry) {
+        return entry.getGlobalAddress() / RECORDS_PER_LOG_FILE;
+    }
+
+    @Override
+    public void append(List<LogData> range) {
+        // Removed trimmed entries
+        List<LogData> entries = new ArrayList<>();
+        for (int x = 0; x < range.size(); x++) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS135)